packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1289,11 +1292,11 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
       }
       case SuspenseComponent: {
         commitSuspenseComponent(finishedWork);
-        attachSuspenseRetryListeners(finishedWork);
+        attachSuspenseRetryListeners(finishedWork, finishedWork.stateNode);
```

Comment:
This isn't the right type. I think you meant to pass `memoizedState` but I think the factoring isn't right here because this function is meant to work with `SuspenseList` too and if it's doesn't, then I'm curious what will break in the `SuspenseList` case.

I think more likely, you'll just want to put your new logic in a separate function that you call from here and not mix it with `attachSuspenseRetryListeners`.

Currently you use the existence of the thennable Set to detect something but I don't think that means what you think it means. It's not a reliable signal for whether something is suspended. I'm not sure what the exact answer is but a good start would be to just move it out here to make it clear that it doesn't have anything to do with whether we attach listeners.